scipy/interpolate/regrid_python.py

```diff
@@ -0,0 +1,1004 @@
+#!/usr/bin/env python3
+import numpy as np
+from scipy.interpolate import BSpline, NdBSpline
+from scipy.interpolate._fitpack_repro import root_rati
+from scipy.sparse import csr_matrix
+import math
+from scipy.sparse import vstack as _vstack, csc_matrix
+from scipy.sparse.linalg import spsolve_triangular as _spsolve_tri
+
+def ndbspline_call_like_bivariate(ndbs, x, y, dx=0, dy=0, grid=True):
+    """
+    Evaluate a 2D NdBSpline using the RectBivariateSpline.__call__ API.
+
+    Parameters
+    ----------
+    ndbs : NdBSpline
+        A 2D NdBSpline object (ndim must be 2).
+    x, y : array_like
+        Input coordinates.
+        If grid is False, x and y will be broadcast to a common shape.
+        If grid is True, x and y should be 1-D, non-decreasing (like FITPACK).
+    dx, dy : int, optional
+        Orders of partial derivatives in x and y (>= 0).
+    grid : bool, optional
+        If True, evaluate on the meshgrid defined by x and y (indexing="ij").
+        If False, evaluate at paired/broadcasted points.
+
+    Returns
+    -------
+    ndarray
+        For grid=True, shape (len(x), len(y)) + trailing_coeff_shape.
+        For grid=False, shape broadcast(x, y).shape + trailing_coeff_shape.
+
+    Notes
+    -----
+    - Extrapolation behavior follows `ndbs.extrapolate`.
+    - Mimics RectBivariateSpline:
+        * grid=True enforces non-decreasing x/y (like the original code).
+        * empty inputs return zeros of the correct dtype/shape.
+    """
+    # Validate: 2D and derivative orders
+    if len(ndbs.t) != 2:
+        raise ValueError("ndbs must be a 2D NdBSpline (len(t) == 2).")
+    if not (isinstance(dx, (int, np.integer)) and isinstance(dy, (int, np.integer))):
+        raise ValueError("dx and dy must be integers.")
+    if dx < 0 or dy < 0:
+        raise ValueError("order of derivative must be positive or zero")
+
+    # Trailing output shape (vector-valued splines supported)
+    trailing = ndbs.c.shape[2:]
+
+    if grid:
+        x = np.asarray(x)
+        y = np.asarray(y)
+
+        # Empty -> zeros (parity with RectBivariateSpline)
+        if x.size == 0 or y.size == 0:
+            return np.zeros((x.size, y.size) + trailing, dtype=ndbs.c.dtype)
+
+        # Monotonic non-decreasing check (same as the original code)
+        if (x.size >= 2) and (not np.all(np.diff(x) >= 0.0)):
+            raise ValueError("x must be strictly increasing when `grid` is True")
+        if (y.size >= 2) and (not np.all(np.diff(y) >= 0.0)):
+            raise ValueError("y must be strictly increasing when `grid` is True")
+
+        # Build ij-grid and stack into (..., 2)
+        X, Y = np.meshgrid(x, y, indexing="ij")
+        xi = np.stack((X, Y), axis=-1)  # (len(x), len(y), 2)
+
+        vals = ndbs(xi, nu=(dx, dy), extrapolate=ndbs.extrapolate)
```

Comment:
`extrapolate=ndbs.extrapolate` should not be needed here. Otherwise it's an NdBSpline bug.